syntax = "proto3";

package image_processor;

////////////////////////////////////////////////////////////////////////////////
//
// Image processing service using gRPC
//
////////////////////////////////////////////////////////////////////////////////

service ImageProcessor
{
  // Apply Gaussian blur to an image
  rpc ApplyGaussianBlur(ImageRequest) returns (ImageResponse);
}

message ImageRequest
{
  bytes image_data = 1;      // Raw image data (JPEG, PNG, etc.)
  int32 kernel_size = 2;     // Gaussian kernel size (default: 9)
  double sigma = 3;          // Gaussian sigma (default: 2.0)
}

message ImageResponse
{
  bytes image_data = 1;      // Processed image data
  bool success = 2;          // Processing status
  string error_message = 3;  // Error message if failed
}
